lib/vagrant/util/directory_changed.rb

```diff
@@ -0,0 +1,23 @@
+require 'pathname'
+
+module Vagrant
+  module Util
+    class DirectoryChanged
+        # Check if directory has any new updates
+        #
+        # @param [Pathname, String] Path to directory
+        # @param [Time] time to compare to eg. has any file in dir_path
+        #               changed since this time
+        # @return [Boolean]
+        def self.directory_changed?(dir_path, threshold_time)
+          dir_path = Pathname.new(dir_path)
+          Dir.glob(dir_path.join("**")).each do |path|
+            if Pathname.new(path).mtime > threshold_time
+              return true
+            end
+          end
+          false
```

Comment:
This can be reduced down to:

```ruby
Dir.glob(Pathname.new(dir_path).join("**", "*")).any? do |path|
  Pathname.new(path).mtime > threshold_time
end
```